pandas/io/pytables.py

```diff
@@ -95,10 +95,10 @@ def _ensure_term(where, scope_level):
                 wlist.append(w)
             else:
                 wlist.append(Term(w, scope_level=level))
-        where = wlist
+        where = wlist if wlist else None
     elif maybe_expression(where):
         where = Term(where, scope_level=level)
-    return where
+    return where if where != "" else None
```

Comment:
you can just do
```where if where else None``` as bool('') and bool([]) evaluate to False

Reply:
I did id at first, but in case some input `where` is some valid expression as string - get after `elif` `where` as `Index`. So this comparsion throws an `The truth value of a Int64Index is ambiguous`.

Reply:
then just do

```
return where if len(where) else None
```